runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -132,7 +133,7 @@ local function diagnostic_vim_to_lsp(diagnostics)
         },
       },
       severity = severity_vim_to_lsp(diagnostic.severity),
-      message = diagnostic.message,
+      message = (diagnostic.user_data and diagnostic.user_data.lsp_original_message) or diagnostic.message,
```

Comment:
This seems out of scope of this PR?

Reply:
Related to https://github.com/neovim/neovim/pull/16286/files/14923f7ce479cd93cddba258261f1e97240b70f0#diff-2c5241b87ba4cc10dd6d1822a2cfa718b1ea523bb9781da4f92585aff7f6b01f, otherwise vim.lsp.diagnostic.get and friends won't return the same diagnostics that were passed in. This is a hack though.